### DISCUSSION TITLE
are there any method to del the text and pic in pdf?

### DISCUSSION THREAD

**Discussion by better319** [pymupdf/PyMuPDF#1696] [Q&A]
are there any method to del the text and pic in pdf?
some examples are needed

in my pdfs  there are may same pics  and same text words . i want to del them。
but i had not found a def to del text or pic.
help needed.
THANKS!!

`
#coding:utf-8
import fitz,os
rooturl='C:\\pdf\\'
url='C:\\pdf\\test.pdf'
doc = fitz.open(url)
imgcount=0
for page in doc: 

      imageList = page.get_images()  
      for imginfo in imageList:
          pix = fitz.Pixmap(doc, imginfo[0])
          pix.save(rooturl+"t_{}.png".format(imgcount))
          print(imginfo[0:10])
          doc._deleteObject(imginfo[0])             '''this seems like use xref,maybe i need a way to compare two pic ???'''
          imgcount+=1
        '''above are  all del,maybe there is a way to del by rect,'''

        areas = page.search_for("613950155")
        for area in areas:
            print(area)
            doc._deleteObject(area[0])    '''error'''

        textlist=page.get_text("613950155")
        for textinfo in textlist:
            print(textinfo)
            doc._deleteObject(textinfo)   '''error'''

doc.save('111.pdf')
`

doc._deleteObject(xref)
pic maybe ok
how to find de text's xref？

**Comment by JorjMcKie** [comment]
>  how to find de text's xref？

Text does not have that!
Use redaction annotations.

**Comment by JorjMcKie** [comment]
There is a detailed example in the utilities repo for removing an image.

**Reply by better319** [reply]
ok,i got it ,but i haven't found the example for del text.
would you like to help me find the url?
THK!

**Reply by JorjMcKie** [reply]
Look up `Page.add_redact_annot()` in the documentation.

**Reply by better319** [reply]
yes, it's a magic work,

`        
r = fitz.Rect(1, 1, 700, 20)

page.add_redact_annot(r, text="", fontname=None, fontsize=11, align=0, fill=(1, 1, 1), text_color=(0, 0, 0), cross_out=True)

page.apply_redactions(1)

`
page.apply_redactions(1)   #1or 0 all the same?

and  the rect was created by myself,

how to get the word's Rect(like:"name:xxxxx")  

page.???    which one return rect?

THK!

**Reply by JorjMcKie** [reply]
You hate reading documentation, don't you 😂?
At least read the tutorial and the recipes please!

Here are some candidates:
* `page.search_for("needle")`
* `page.get_text("words")` then search for the word you like

Other text extraction variants also give you the font name and size and color or properties like bold, italic, ...

**Comment by JorjMcKie** [comment]
... and of course all that is no bug ... not even an issue, therefore I moved this to Discussions.
